src/base/InstanceId.h

```diff
@@ -49,6 +49,7 @@ class InstanceId
     typedef ValueType Value; ///< id storage type
 
     InstanceId() {change();}
+    InstanceId(const InstanceId &); ///< no copying; IDs are unique
```

Comment:
AFAICT, this PR change starts allowing InstanceId copying by assignment. There are two ways to fix that, depending on whether we want to (continue to) ban moving:

To allow[^1] moving InstanceId objects by explicitly declaring default move constructor and move assignment operator:

```suggestion
    // no copying (because IDs are unique) but moving is OK
    InstanceId(const InstanceId &) = delete;
    InstanceId &operator=(const InstanceId &) = delete;
    InstanceId(InstanceId &&) = defailt;
    InstanceId &operator=(InstanceId &&) = default;
```

To (continue to) prohibit moving InstanceId objects, use this idiomatic shorter variant that prohibits everything at once:

```suggestion
    InstanceId(InstanceId &&) = delete; // no copying/moving of any kind
```


[^1]: A user-declared copy constructor inhibits move members. See page 25+ of the well-known "Special Members" [slides](https://accu.org/conf-docs/PDFs_2014/Howard_Hinnant_Accu_2014.pdf).